code/parachain/frame/pallet-multihop-xcm-ibc/src/lib.rs

```diff
@@ -578,6 +648,9 @@ pub mod pallet {
 			let result = pallet_ibc::Pallet::<T>::transfer(
 				signed_account_id.into(),
 				transfer_params,
+
+				// please fix unwraps as per `cargo clippy --package pallet-multihop-xcm-ibc`
+				// `.unwrap()` - with let should work nice
 				asset_id.unwrap(), //TODO remove unwrap
```

Comment:
Why is asset_id even an Option?

Reply:
yeah, please check all issues on top of function with ?(residual), so that algorithm is clean later